clang/lib/Analysis/FlowSensitive/ControlFlowContext.cpp

```diff
@@ -94,6 +94,36 @@ static llvm::BitVector findReachableBlocks(const CFG &Cfg) {
   return BlockReachable;
 }
 
+static llvm::DenseSet<const CFGBlock *>
+buildContainsExprConsumedInDifferentBlock(
+    const CFG &Cfg,
+    const llvm::DenseMap<const Stmt *, const CFGBlock *> &StmtToBlock) {
+  llvm::DenseSet<const CFGBlock *> Result;
+
+  auto CheckChildExprs = [&Result, &StmtToBlock](const Stmt *S,
+                                                 const CFGBlock *Block) {
+    for (const Stmt *Child : S->children()) {
+      const CFGBlock *ChildBlock = StmtToBlock.lookup(Child);
+      if (isa<Expr>(Child) && ChildBlock != Block)
```

Comment:
nit: check `isa<Expr>` before the lookup? avoids the extra lookup when it's not an expression.

Reply:
Good point -- done!